app/models/miq_report_result.rb

```diff
@@ -70,21 +70,19 @@ def miq_group_description
 
   def report_results
     if binary_blob
-      serializer_name = binary_blob.data_type
-      serializer_name = "Marshal" unless serializer_name == "YAML"  # YAML or Marshal, for now
-      serializer = serializer_name.constantize
-      loaded = serializer.load(binary_blob.binary)
+      loaded = binary_blob.load
       loaded.kind_of?(Hash) ? MiqReport.from_hash(loaded) : loaded
     elsif report.kind_of?(MiqReport)
       report
-    else
-      nil
     end
   end
 
   def report_results=(value)
-    self.binary_blob = BinaryBlob.new(:name => "report_results", :data_type => "YAML")
-    binary_blob.binary = YAML.dump(value.kind_of?(MiqReport) ? value.to_hash : value)
+    build_binary_blob(
+      :name      => "report_results",
+      :data_type => "YAML",
+      :binary    => YAML.dump(value.kind_of?(MiqReport) ? value.to_hash : value)
```

Comment:
Yeah... it doesn't seem hugely *likely* to change, but I certainly wouldn't consider it contracted.

Could explicitly pull that attribute out in a custom `initialize`?